Classes/Queue/QueueInterface.php

```diff
@@ -37,17 +37,17 @@ public function getName();
      * @param array $options Simple key/value array with options, supported options depend on the queue implementation
      * @return string The identifier of the message under which it was queued
      */
-    public function submit($payload, array $options = []);
+    public function submit($payload, array $options = []): string;
 
     /**
      * Wait for a message in the queue and remove the message from the queue for processing
      * If a non-null value was returned, the message was not queued. Otherwise a timeout
      * occurred and no message was available or received.
      *
-     * @param integer $timeout
+     * @param int $timeout
      * @return Message The received message or NULL if a timeout occurred
      */
-    public function waitAndTake($timeout = null);
+    public function waitAndTake(?int $timeout = null): ?Message;
```

Comment:
This will require PHP 7.1+ which is fine I guess. But we should maybe add it to the composer dependencies

Reply:
done with 322ec0c735ec665d6ae026bdca94ee5ad8891e80